src/scala/scripts/TwitterScroogeGenerator.scala

```diff
@@ -0,0 +1,269 @@
+      if (file.isDirectory()) f.mkdir()
+      else {
+        val path = Paths.get(f.toURI)
+
+        val is = jar.getInputStream(file)
+        try {
+          Files.copy(is, path) // Will error out if path already exists
+        } finally {
+          is.close()
+        }
+
+        files += path
+      }
+    }
+    files.toList
+  }
+
+  //TODO add logging that can be turned on/off?
+  def main(args: Array[String]) {
+    if (args.length < 4) sys.error("Need to ensure enough arguments! " +
+      "Required 3 arguments: onlyTransitiveThriftSrcs immediateThriftSrcs " +
+      "jarOutput remoteJarsFile. Received: " + args.mkString(","))
+
+    val onlyTransitiveThriftSrcsFile = args(0)
+    val immediateThriftSrcsFile = args(1)
+    val jarOutput = args(2)
+    val remoteJarsFile = args(3)
+
+    val tmp = Paths.get(Option(System.getenv("TMPDIR")).getOrElse("/tmp"))
+    val scroogeOutput = Files.createTempDirectory(tmp, "scrooge").toString
+
+    // These are all of the files to include when generating scrooge
+    // Should not include anything in immediateThriftSrcs
+    val onlyTransitiveThriftSrcJars =
+      Source.fromFile(onlyTransitiveThriftSrcsFile).getLines.toSet
+
+    // These are the files whose output we want
+    val immediateThriftSrcJars =
+      Source.fromFile(immediateThriftSrcsFile).getLines.toSet
+
+    val genFileMap = s"$scroogeOutput/gen-file-map.txt"
+
+    val scrooge = new Compiler
+
+    // we need to extract into the same tree, as that is the only way to get relative imports between them working..
+    // AS SUCH, we are just going to try extracting EVERYTHING to the same tree, and we will just error if there
+    // are more than one.
+    val _tmp =  Files.createTempDirectory(tmp, "jar")
+    // This will only be meaningful if they have absolute_prefix set
+    scrooge.includePaths += _tmp.toString
+
+    //TODO we should probably just make everything Paths instead of strings
+    def extract(jars: Set[String]): Set[String] =
+      jars.flatMap { jar =>
+        val files = extractJarTo(jar, _tmp.toString)
+        files.foreach { scrooge.includePaths += _.toString }
+        files.map(_.toString)
+      }
+
+    val immediateThriftSrcs = extract(immediateThriftSrcJars)
+
+    immediateThriftSrcs.foreach { scrooge.thriftFiles += _ }
+
+    val onlyTransitiveThriftSrcs = extract(onlyTransitiveThriftSrcJars)
+
+    val intersect = onlyTransitiveThriftSrcs.intersect(immediateThriftSrcs)
+
+    if (intersect.nonEmpty)
+      sys.error("onlyTransitiveThriftSrcs and immediateThriftSrcs should " +
+        s"have not intersection, found: ${intersect.mkString(",")}")
+
+    //TODO WE NEED TO TEST THIS!!
+    val remoteSrcJars = Source.fromFile(remoteJarsFile).getLines.toSet
+    extract(remoteSrcJars)
+
+    val dirsToDelete =
+      Set(
+        immediateThriftSrcsFile,
+        onlyTransitiveThriftSrcsFile,
+        scroogeOutput,
+        remoteJarsFile,
```

Comment:
we should not be deleting the inputs. Only delete the tmp directory and the scroogeOutput.